doc/modules/linear_model.rst

```diff
@@ -860,28 +860,71 @@ regularization.
     that it improves numerical stability. No regularization amounts to
     setting C to a very high value.
 
-As an optimization problem, binary class :math:`\ell_2` penalized logistic
-regression minimizes the following cost function:
+Binary Case
+-----------
 
-.. math:: \min_{w, c} \frac{1}{2}w^T w + C \sum_{i=1}^n \log(\exp(- y_i (X_i^T w + c)) + 1) .
+For notational ease, we assume that the target :math:`y_i` takes values in the
+set :math:`\{-1, 1\}` at trial :math:`i`. As an optimization problem, binary
+class logistic regression using :math:`r(w)` regularization minimizes the
+following cost function:
 
-Similarly, :math:`\ell_1` regularized logistic regression solves the following
+.. math:: \min_{w, c} r(w) + C \sum_{i=1}^n \log(\exp(- y_i (X_i^T w + c)) + 1) .
+
+
+Multinomial Case
+----------------
+
+We may then extend logistic regression to obtain a multinomial estimator by
+considering the logistic regression as a `log-linear model
+<https://en.wikipedia.org/wiki/Multinomial_logistic_regression#As_a_log-linear_model>`_.
+
+.. note::
+   It is possible in a :math:`K`-class context to parameterize the model
+   using only :math:`K-1` weight vectors, leaving one class probability fully
+   determined by the other class probabilities by leveraging the fact that all
+   class probabilities must sum to one. We choose to overparameterize the model
+   using :math:`K` weight vectors for ease of implementation and to preserve the
+   symmetrical inductive bias regarding ordering of classes. This effect becomes
+   especially important when using regularization.
+
+Let :math:`J_i` be a binary vector with a :math:`0` for every element except
+for element :math:`i`. In the multinomial context with :math:`K`-many classes,
+we define the target vector of :math:`X_n` as :math:`Y_n=J_t` where :math:`t`
+is the true class of :math:`X_n`. Instead of a single weight vector, we now have
+a matrix of weights :math:`W` where each vector :math:`W_k` corresponds to class
+:math:`k`. Then we can define the evidence vector :math:`z_n` component-wise as:
+
+.. math:: p(Y_n=J_k|X_n) = z_{n,k} = \frac{\exp (W_k^T X_n)}{\sum_j \exp (W_j^T X_n)}
```

Comment:
```suggestion
Let :math:`y_i \in {0, \ldots, K-1}` be the label (ordinal) encoded target variable for observation $i$.
Instead of a single coefficient vector, we now have
a matrix of coefficients :math:`W` where each row vector :math:`w_k` corresponds to class
:math:`k`. We aim at predicting the class probabilities via ``predict_proba`` as:

.. math:: p(y_i=k|X_i) = \frac{\exp(X_i w_k + w_{0, k})}{\sum_{l=0}^{K-1} \exp(X_i w_l + w_{0, l})}
```